### ISSUE TITLE
An errored frontend is not removed when you leave its route.

### ISSUE THREAD

**Issue by victorlevasseur** [single-spa/single-spa-layout#105]
I have the given configuration : 
```html
<template id="single-spa-layout">
    <single-spa-router>
      <nav>
        <application name="@onepoc/menu" error="menu"></application>
      </nav>
      <main>
        <route path="auth">
          <application loader="spinner" error="main" name="@onepoc/auth"></application>
        </route>
        <route path="gift">
          <application loader="spinner" error="main" name="@onepoc/gift"></application>
        </route>
        <route path="pro">
          <application loader="spinner" error="main" name="@onepoc/pro"></application>
        </route>
        <route path="test">
          <application loader="spinner" error="main" name="@onepoc/test"></application>
        </route>
        <route default>
          <div class="error-page"><p>Oups ! Vous êtes nulle part !<br/><small>C'est pas tout à fait pro !</small></p></div>
        </route>
      </main>
    </single-spa-router>
  </template>
```

The menu (@onepoc/menu) is always displayed. We use it to navigate to the different routes.
The other frontends do not exist yet, so their error handler is displayed.

However, when I navigate to another route (for example from /gift to /pro), the "gift" error handler stays on the page and the "pro" error handler appears.

I expect the "pro" error handler to appear but the "gift" error handler to dissappear.

# Reproduction
Here is our proof of concept project on which the problem can be reproduced.

https://d15b5tf1uf3fd3.cloudfront.net/gift

By starting on the "/gift" prefix, you'll see the "gift" error handler. Then, you can try to navigate to the "P" link in the side menu, you'll see the second error handler appearing and (unfortunately) the first one staying.

**Comment by victorlevasseur** [comment]
I'm not quite sure it's related to #103.

**Comment by joeldenning** [comment]
Thanks for the clear description, I've identified the problem and am working on a fix for it.

**Comment by joeldenning** [comment]
I've put together a fix for this at https://github.com/single-spa/single-spa-layout/pull/106

**Comment by joeldenning** [comment]
The fix for this is released in https://github.com/single-spa/single-spa-layout/releases/tag/v1.4.0

**Comment by vmizg** [comment]
Hi @joeldenning,
I've come across a very similar issue and I wonder if it could be related, although what's bizarre is that even after the fix you introduced in 1.4.0 I'm still seeing it happen (on 1.5.3).

Here's a visual description of the issue:

![singlespa-weird](https://user-images.githubusercontent.com/9007308/122349853-39a76500-cf7f-11eb-9733-970c84cf15ad.png)

Basically, if an MFE throws an error, we've set up a parcel inside layout data that we pass to `constructRoutes` and register the parcel with the corresponding mfe in HTML. This parcel is then rendered if that MFE throws.

But the problem is that, upon navigation to `@mfe/two`, while single spa imports the relevant MFE, *both* the loading parcel for the new mfe and the error parcel for the previous one are displayed. The URL has already changed to the path of @mfe/two at that point.

Is this a bug or could I be missing some configuration there?

**Comment by joeldenning** [comment]
@vmizg this sounds like a bug. Could you create a new github issue with a demonstration of the issue?
